lib/kafka/protocol/fetch_request.rb

```diff
@@ -46,6 +48,7 @@ def encode(encoder)
         encoder.write_int32(@max_wait_time)
         encoder.write_int32(@min_bytes)
         encoder.write_int32(@max_bytes)
+        encoder.write_int8(ISOLATION_READ_COMMITTED)
```

Comment:
@nguyenquangminh0711 : Sorry, I know this change is ancient history, but do you happen to recall why this is hard-coded to `ISOLATION_READ_COMMITTED` when it looks like there was some intent to make this configurable (given that `ISOLATION_READ_UNCOMMITTED` is defined?) I have a situation where I need to be able to read uncommitted messages, but I want to make sure there isn't some good reason for not having this be configurable before I go down that path.

Reply:
@jwilger This hard-coding was for the sake of simplicity. This implementation was complicated enough that it would be hard to ensure all the configurations worked well. Therefore, I wanted to focus on read committed first, then implemented the custom configuration to allow semantic configuration later. However,  I never had a use case for Read uncommitted, and nobody asked about this until now. So, I think it's feasible to add a configuration for this.

Reply:
Perfect, thank you! We will likely be submitting a PR for it soon-ish (for the moment, we just forked ruby-kafka and flip-flopped the hard-coding, because we needed to get it working quickly due to a stuck partition. N.B. it's no fun when your kafka cluster gets restarted in the middle of a transaction.)